File: client-scalaz/src/main/scala/org/openapitools/client/api/ShipmentOrder.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime

import java.time.OffsetDateTime

import ShipmentOrder._

case class ShipmentOrder (
  valid: Option[Boolean],
batchID: Option[Integer],
batchSize: Option[Integer],
orderTrackingID: Option[Double],
clientRefNo: Option[String],
clientRefNo2: Option[String],
hubCode: Option[String],
accountNo: Option[String],
companyName: Option[String],
vehicleTypeName: Option[String],
serviceTypeName: Option[String],
serviceCode: Option[String],
pieces: Option[Integer],
weight: Option[Double],
cubicFeet: Option[Double],
pickupCompany: Option[String],
pickupContact: Option[String],
pickupPhone: Option[String],
pickupStreet: Option[String],
pickupStreet2: Option[String],
pickupCity: Option[String],
pickupState: Option[String],
pickupZip: Option[String],
pickupTargetFrom: Option[OffsetDateTime],
pickupTargetTo: Option[OffsetDateTime],
deliveryCompany: Option[String],
deliveryContact: Option[String],
deliveryPhone: Option[String],
deliveryStreet: Option[String],
deliveryStreet2: Option[String],
deliveryCity: Option[String],
deliveryState: Option[String],
deliveryZip: Option[String],
deliveryTargetFrom: Option[OffsetDateTime],
deliveryTargetTo: Option[OffsetDateTime],
orderPackages: Option[List[OrderPackage]],
routeID: Option[Integer],
driverID: Option[Integer],
pickupSequenceNo: Option[Integer],
deliverySequenceNo: Option[Integer],
pickupETA: Option[String],
deliveryETA: Option[String])

object ShipmentOrder {
  import DateTimeCodecs._

  implicit val ShipmentOrderCodecJson: CodecJson[ShipmentOrder] = CodecJson.derive[ShipmentOrder]
  implicit val ShipmentOrderDecoder: EntityDecoder[ShipmentOrder] = jsonOf[ShipmentOrder]
  implicit val ShipmentOrderEncoder: EntityEncoder[ShipmentOrder] = jsonEncoderOf[ShipmentOrder]
}

File: client-clojure/src/sirqul_io_t_platform/specs/shipment_order.clj
(ns sirqul-io-t-platform.specs.shipment-order
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.order-package :refer :all]
            )
  (:import (java.io File)))


(def shipment-order-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :BatchID) int?
   (ds/opt :BatchSize) int?
   (ds/opt :OrderTrackingID) float?
   (ds/opt :ClientRefNo) string?
   (ds/opt :ClientRefNo2) string?
   (ds/opt :HubCode) string?
   (ds/opt :AccountNo) string?
   (ds/opt :CompanyName) string?
   (ds/opt :VehicleTypeName) string?
   (ds/opt :ServiceTypeName) string?
   (ds/opt :ServiceCode) string?
   (ds/opt :Pieces) int?
   (ds/opt :Weight) float?
   (ds/opt :CubicFeet) float?
   (ds/opt :PickupCompany) string?
   (ds/opt :PickupContact) string?
   (ds/opt :PickupPhone) string?
   (ds/opt :PickupStreet) string?
   (ds/opt :PickupStreet2) string?
   (ds/opt :PickupCity) string?
   (ds/opt :PickupState) string?
   (ds/opt :PickupZip) string?
   (ds/opt :PickupTargetFrom) inst?
   (ds/opt :PickupTargetTo) inst?
   (ds/opt :DeliveryCompany) string?
   (ds/opt :DeliveryContact) string?
   (ds/opt :DeliveryPhone) string?
   (ds/opt :DeliveryStreet) string?
   (ds/opt :DeliveryStreet2) string?
   (ds/opt :DeliveryCity) string?
   (ds/opt :DeliveryState) string?
   (ds/opt :DeliveryZip) string?
   (ds/opt :DeliveryTargetFrom) inst?
   (ds/opt :DeliveryTargetTo) inst?
   (ds/opt :OrderPackages) (s/coll-of order-package-spec)
   (ds/opt :RouteID) int?
   (ds/opt :DriverID) int?
   (ds/opt :PickupSequenceNo) int?
   (ds/opt :DeliverySequenceNo) int?
   (ds/opt :PickupETA) string?
   (ds/opt :DeliveryETA) string?
   })

(def shipment-order-spec
  (ds/spec
    {:name ::shipment-order
     :spec shipment-order-data}))
